test/Concurrency/attr_execution_conversions.swift

```diff
@@ -77,3 +77,59 @@ do {
   // expected-error@-1 {{cannot convert value of type '(@execution(caller) () async -> ()).Type' to expected argument type '(@isolated(any) () async -> Void).Type'}}
 }
 
+
+// Converting to `@execution(caller)` function
+class NonSendable {}
+
+func testNonSendableDiagnostics(
+  globalActor1: @escaping @Sendable @MainActor (NonSendable) async -> Void,
+  globalActor2: @escaping @Sendable @MainActor () async -> NonSendable,
+  erased1: @escaping @Sendable @isolated(any) (NonSendable) async -> Void,
+  erased2: @escaping @Sendable @isolated(any) () async -> NonSendable,
+  nonIsolated1: @escaping @Sendable (NonSendable) -> Void,
+  nonIsolated2: @escaping @Sendable @execution(concurrent) (NonSendable) async -> Void,
+  nonIsolated3: @escaping @Sendable () -> NonSendable,
+  nonIsolated4: @escaping @Sendable @execution(concurrent) () async -> NonSendable,
+  caller1: @escaping @Sendable @execution(caller) (NonSendable) async -> Void,
+  caller2: @escaping @Sendable @execution(caller) () async -> NonSendable
+) {
+  let _: @execution(caller) (NonSendable) async -> Void = globalActor1 // expected-note {{type 'NonSendable' does not conform to 'Sendable' protocol}}
+  // expected-error@-1 {{cannot convert '@MainActor @Sendable (NonSendable) async -> Void' to '@execution(caller) (NonSendable) async -> Void' because crossing of an isolation boundary requires parameter and result types to conform to 'Sendable' protocol}}
+  let _: @execution(caller) () async -> NonSendable = globalActor2 // expected-note {{type 'NonSendable' does not conform to 'Sendable' protocol}}
+  // expected-error@-1 {{cannot convert '@MainActor @Sendable () async -> NonSendable' to '@execution(caller) () async -> NonSendable' because crossing of an isolation boundary requires parameter and result types to conform to 'Sendable' protocol}}
+
+  let _: @execution(caller) (NonSendable) async -> Void = erased1 // expected-note {{type 'NonSendable' does not conform to 'Sendable' protocol}}
+  // expected-error@-1 {{cannot convert '@isolated(any) @Sendable (NonSendable) async -> Void' to '@execution(caller) (NonSendable) async -> Void' because crossing of an isolation boundary requires parameter and result types to conform to 'Sendable' protocol}}
+  let _: @execution(caller) () async -> NonSendable = erased2 // expected-note {{type 'NonSendable' does not conform to 'Sendable' protocol}}
+  // expected-error@-1 {{cannot convert '@isolated(any) @Sendable () async -> NonSendable' to '@execution(caller) () async -> NonSendable' because crossing of an isolation boundary requires parameter and result types to conform to 'Sendable' protocol}}
+
+  let _: @execution(caller) (NonSendable) async -> Void = nonIsolated1 // expected-note {{type 'NonSendable' does not conform to 'Sendable' protocol}}
```

Comment:
I think this conversion is safe. `nonIsolated1` is synchronous and nonisolated, so it always runs where it's called from, and that's fine if the destination type is `@execution(caller)`.

Reply:
Yes, thank you!